go/api/handlers/photo.go

```diff
@@ -0,0 +1,185 @@
+		return
+	}
+
+	filePath, errFilePath := utils.SaveUploadedFile(file, "photo", title)
+	if errFilePath != nil {
+		logger.ApiError(context, "Error saving file: "+filePath)
+		context.JSON(http.StatusInternalServerError, gin.H{"error": "Error saving file"})
+		return
+	}
+
+	var photo = models.Photo{
+		Title:       title,
+		Description: description,
+		TripID:      uint(tripId),
+		Path:        filePath,
+	}
+
+	err := handler.Repository.Create(&photo)
+	if err != nil {
+		errorHandlers.HandleGormErrors(err, context)
+		return
+	}
+
+	photoResponse := responses.PhotoResponse{
+		ID:          photo.ID,
+		Title:       photo.Title,
+		Description: photo.Description,
+		Path:        photo.Path,
+		CreatedAt:   photo.CreatedAt.Format(time.RFC3339),
+		UpdateAt:    photo.UpdatedAt.Format(time.RFC3339),
+	}
+
+	context.JSON(200, photoResponse)
+	logger.ApiInfo(context, "Create a photo "+photo.Path+" on trip "+tripIdStr)
+}
+
+// @Summary      Download a photo
+// @Description  Download a photo
+// @Tags         photo
+// @Accept       json
+// @Produce      application/octet-stream
+// @Security     BearerAuth
+// @Param        id          path      string  true  "ID of the trip"
+// @Param        photoID  path      string  true  "ID of the photo"
+// @Success      200         {file}    true    "A file stream of the photo"
+// @Failure      400         {object}  error
+// @Failure      401         {object}  error
+// @Failure      404         {object}  error
+// @Router       /trips/{id}/photos/{photoID}/download [get]
+func (handler *PhotoHandler) DownloadPhoto(context *gin.Context) {
+	photoID := context.Param("photoID")
+
+	photo, errPhoto := handler.Repository.Get(photoID)
+	if errPhoto != nil {
+		errorHandlers.HandleGormErrors(errPhoto, context)
+		return
+	}
+
+	filepath, errFilePath := utils.GetFilePath("photo", photo.Path)
+	if errFilePath != nil {
+		logger.ApiError(context, "Failed to get file path "+photo.Path)
+		context.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to get file path"})
+		return
+	}
+	context.Header("Content-Disposition", "attachment; filename=\""+photo.Path+"\"")
+	context.File(filepath)
+	logger.ApiInfo(context, "Download a photo "+photo.Path)
+}
+
+// @Summary		Delete a photo from trip
+// @Description	Delete a photo from the trip
+// @Tags			photo
+// @Accept			json
+// @Produce		json
+// @Security		BearerAuth
+// @Param			id			path		string	true	"ID of the trip"
+// @Param			photoID	path		string	true	"ID of the photo"
+// @Success		204			{object}	error
+// @Failure		400			{object}	error
+// @Failure		401			{object}	error
+// @Failure		404			{object}	error
+// @Router			/trips/{id}/photos/{photoID} [delete]
+func (handler *PhotoHandler) DeletePhotoFromTrip(context *gin.Context) {
+	photoID := context.Param("photoID")
+	if photoID == "" {
+		logger.ApiWarning(context, "No photo ID provided")
+		context.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
```

Comment:
```suggestion
	}
	// ICI
```

Pareil, il faut vérifier que la photo appartient bien au trip et que le mec est bien l'uploader de la photo